js/script.js

```diff
@@ -0,0 +1,155 @@
+                names.appendChild(namestxt);
+
+                ratingcontainer = document.createElement("div");
+                ratingcontainer.className = "rating";
+                
+                rate = document.createElement("span");     
+                ratetxt = document.createTextNode(data.record[course].courses[item].rating.toFixed(2) + " "); 
+                rate.appendChild(ratetxt);
+
+                star1 = document.createElement("span");
+                star1.className = "fa fa-star";
+                star2 = document.createElement("span");
+                star2.className = "fa fa-star";
+                star3 = document.createElement("span");
+                star3.className = "fa fa-star";
+                star4 = document.createElement("span");
+                star4.className = "fa fa-star";
+                star5 = document.createElement("span");
+                star5.className = "fa fa-star-half-stroke";
+                ratingcontainer.appendChild(rate);
+                ratingcontainer.appendChild(star1);
+                ratingcontainer.appendChild(star2);
+                ratingcontainer.appendChild(star3);
+                ratingcontainer.appendChild(star4);
+                ratingcontainer.appendChild(star5);
+
+                price = document.createElement("span")
+                pricetxt = document.createTextNode('$' + data.record[course].courses[item].price)
+                price.appendChild(pricetxt);
+
+                datacontainer.appendChild(image);
+                datacontainer.appendChild(heading);
+                datacontainer.appendChild(names);
+                datacontainer.appendChild(ratingcontainer);
+                datacontainer.appendChild(price);
+
+                cardlink.appendChild(datacontainer);
+
+
+                cards_container.appendChild(cardlink);
+            }
+        }
+    }
+    section.appendChild(cards_container);
+}
+
+
+// change the courses when change the topic
+for(const item in courses){
+    const course = document.getElementById(courses[item]);
+    course.addEventListener("change", function(){
+        cur_course = course.id;
+        build_description(cur_course);
+    })
+}
+
+// building searching description
+function build_search(searchvalue){
+    section.innerHTML = "";
+    const description = `
+    <h2>Your search results for "${searchvalue}" are: </h2>
+    <br><br>
+    `
+    section.innerHTML += description;
+    build_cards("", searchvalue);
+}
+
+// search on submit in search bar
+document.forms[0].onsubmit = function(e){
+    e.preventDefault();
+    let searchvalue = document.querySelector("[class = 'search-field']").value;
+    let x = searchvalue.split(' ').length - 1;
+    // prevent searching on empty or only spaces values 
+    if(searchvalue !== '' &&  x !== searchvalue.length){
+        build_search(searchvalue);
+    }
+}   
+
+// featch the data on loading 
+fetchdata();
```

Comment:
add a blank line at the end of the file.
```suggestion
fetchdata();

```